From abf23253a574e02aa243fd91e4e754144c160516 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Mon, 19 Jul 2021 15:31:16 +0800
Subject: [PATCH] drivers: pci: modify the config judgement to fix build errors

There are some modifications of pcie driver for s32gen1 platform, so
modify the config judgement accordingly to fix build errors, like:

	error: implicit declaration of function 's32gen1_pcie_config_irq'; did you mean 's32gen1_pcie_config_common'? [-Werror=implicit-function-declaration]
	|  1289 |                 ret = s32gen1_pcie_config_irq(&pp->msi_irq, "msi", pdev,

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/controller/dwc/pci-s32gen1.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-s32gen1.c b/drivers/pci/controller/dwc/pci-s32gen1.c
index fb2fe809040f..a1c2092934c5 100644
--- a/drivers/pci/controller/dwc/pci-s32gen1.c
+++ b/drivers/pci/controller/dwc/pci-s32gen1.c
@@ -611,7 +611,7 @@ static void s32gen1_pcie_stop_link(struct dw_pcie *pcie)
 	s32gen1_pcie_disable_ltssm(s32_pp);
 }
 
-#ifdef CONFIG_PCI_S32GEN1_EP_MSI
+#ifdef CONFIG_PCI_MSI
 /* msi IRQ handler
  * irq - interrupt number
  * arg - pointer to the "struct pcie_port" object
@@ -628,7 +628,7 @@ static irqreturn_t s32gen1_pcie_msi_handler(int irq, void *arg)
 
 	return dw_handle_msi_irq(pp);
 }
-#endif /* CONFIG_PCI_S32GEN1_EP_MSI */
+#endif /* CONFIG_PCI_MSI */
 
 static int s32gen1_pcie_host_init(struct pcie_port *pp)
 {
@@ -685,7 +685,7 @@ static struct dw_pcie_host_ops s32gen1_pcie_host_ops2 = {
 	.msi_host_init = s32gen1_pcie_msi_host_init,
 };
 
-#if defined(CONFIG_PCI_S32GEN1_EP_MSI) || defined(CONFIG_PCI_DW_DMA)
+#if defined(CONFIG_PCI_MSI) || defined(CONFIG_PCI_DW_DMA)
 #define MAX_IRQ_NAME_SIZE 32
 static int s32gen1_pcie_config_irq(int *irq_id, char *irq_name,
 		struct platform_device *pdev,
-- 
2.25.1

